content/docs/buildpack-author-guide/create-buildpack/build-app.md

```diff
@@ -19,20 +30,44 @@ echo "---> Ruby Buildpack"
 # 1. GET ARGS
 layersdir=$1
 
-# 2. DOWNLOAD RUBY
-echo "---> Downloading and extracting Ruby"
+# 2. CREATE THE LAYER DIRECTORY
 rubylayer="$layersdir"/ruby
 mkdir -p "$rubylayer"
+```
+
+The `rubylayer` directory is a sub-directory of the directory provided as the first positional argument to the script (the [layers directory][layers-dir]), and this is where we'll store the Ruby runtime.
+
+Next, we'll download the Ruby runtime and install it into the layer directory. Add the following code to the end of the `build` script:
+
+```bash
+# 3. DOWNLOAD RUBY
+echo "---> Downloading and extracting Ruby"
 ruby_url=https://s3-external-1.amazonaws.com/heroku-buildpack-ruby/heroku-18/ruby-2.5.1.tgz
 wget -q -O - "$ruby_url" | tar -xzf - -C "$rubylayer"
+```
 
-# 3. MAKE RUBY AVAILABLE DURING LAUNCH
+This code uses the `wget` tool to download the Ruby binaries from the given URL, and extracts it to the `rubylayer` directory.
+
+The last step in creating a layer is writing a TOML file that contains metadata about the layer. Without this file, the Buildpack lifecycle will ignore the layer directory. For the Ruby layer, we need to ensure it is available in the launch image by setting the `launch` key to `true`. Add the following code to script:
+
+```bash
+# 4. MAKE RUBY AVAILABLE DURING LAUNCH
 echo -e 'launch = true' > "$rubylayer.toml"
+```
+
+### Installing Dependencies
+
+Next, we'll use the Ruby runtime you installed to download the application's dependencies. First, we need to make the Ruby executables available to our script by putting it on the Path. Add the following code to the end of the `build` script:
 
+```bash
 # 4. MAKE RUBY AVAILABLE TO THIS SCRIPT
 export PATH="$rubylayer"/bin:$PATH
 export LD_LIBRARY_PATH=${LD_LIBRARY_PATH:+${LD_LIBRARY_PATH}:}"$rubylayer/lib"
+```
+
+Now we can install Bundler, a dependency manager for Ruby, and run the `bundle install` command. Append the following code to the script:
 
+```bash
```

Comment:
Maybe it's worth calling out where the result of these commands end up - i.e., in the ruby directory and/or workspace. 

This is a bit more involved, but I feel like this tutorial could benefit from a `dive test-ruby-app` mapping each of the commands to a layer in the final image.

Reply:
I'm not sure I follow, but if that's an enhancement to `build` script, I'd rather put that in a separate PR. I was mostly hoping to reorganize this so it explains each step (and has an anchor link for creating a layer)

Reply:
What I understood from @natalieparellano's comment was that it may be helpful to add a visual on how each set of commands translate to layers on the final image.

ie. similar to what dive displays

![image](https://user-images.githubusercontent.com/475559/107545171-9b4e6180-6b90-11eb-8657-5911fb83a68c.png)